manager/controlapi/service.go

```diff
@@ -21,6 +22,9 @@ var (
 	errModeChangeNotAllowed      = errors.New("service mode change is not allowed")
 )
 
+// Regexp pattern for hostname to conform RFC 1123
+var hostnamePattern = regexp.MustCompile("^(([[:alnum:]]|[[:alnum:]][[:alnum:]\\-]*[[:alnum:]])\\.)*([[:alnum:]]|[[:alnum:]][[:alnum:]\\-]*[[:alnum:]])$")
```

Comment:
@vasil-yordanov Let's see if we can get a unit test with some examples for this.